test/unit/connection/ClientMessageReaderTestRenewed.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+const { expect } = require('chai');
+
+const { ClientMessageReader } = require('../../../lib/network/Connection');
+const cm = require('../../../lib/protocol/ClientMessage');
+const {ClientMessage, Frame} = require("../../../lib/protocol/ClientMessage");
+
+describe('ClientMessageReaderTest', function () {
+
+    const HEADER_SIZE = cm.SIZE_OF_FRAME_LENGTH_AND_FLAGS;
+
+    function createFrameWithRandomBytes(length) {
+        const buffer = Buffer.allocUnsafe(length);
+        for (let i = 0; i < length; i++) {
+            buffer.writeInt8(Math.floor(Math.random()), i);
+        }
+        return new Frame(buffer);
+    }
+
+    function writeToBuffer(message) {
+        const length = message.getTotalLength() + HEADER_SIZE;
+        const buffer = Buffer.allocUnsafe(length);
+        message.writeTo(buffer);
+        return buffer;
+    }
+
+    let reader;
+    beforeEach(function() {
+        reader = new ClientMessageReader();
+    });
+
+    it('reads single message', function() {
```

Comment:
I named all of them as Javas but it seems weird now with the old test cases of Nodejs. Can you look it again? Renaming new cases would be better I think

Reply:
The test naming convention is here https://hazelcast.atlassian.net/wiki/spaces/HZC/pages/4048027654/Test+Porting+From+Java+Client+Tests+and+Naming+Convention

So please keep them as same with Java test names and also if you can add the reference of the java test as a comment it would be great. You can get it via `Copy/Paste Special / Copy Reference` in Intellij IDEA IDE. it produces something like `com.hazelcast.client.impl.protocol.util.ClientMessageReaderTest#testReadSingleFrameMessage`